lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,171 @@
+        )
+
+        # Retrieve the ES indices that belong to this run_id from the metadata
+        # table
+        indices = self.get_index(dataset, "run-toc")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix run-toc"
+                f" for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        return {
+            "path": f"/{indices}/_search",
+            "kwargs": {
+                "json": {
+                    "size": 10000,
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {"match": {"run_data_parent": run_id}},
+                                {"regexp": {"directory": f"{parent}[^/]+"}},
+                            ],
+                            "must_not": {
+                                "regexp": {"directory": f"{parent}[^/]+/[^/]+"}
+                            },
+                        }
+                    },
+                }
+            },
+        }
+
+    def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
+        """
+        Returns a Flask Response containing a JSON object (keyword/value
+        pairs) with directories and files each of which is a list of
+        entries describing individual files and directories.
+
+        Example:
+
+            {
+                "_index": "riya-pbench.v6.run-toc.2021-05",
+                "_type": "_doc",
+                "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                "_score": 0.0,
+                "_source": {
+                    "parent": "/default",
+                    "directory": "/default/sample",
+                    "mtime": "2021-05-21T24:00:00",
+                    "mode": "0o755",
+                    "name": "sample",
+                    "ancestor_path_elements": ["default"],
+                    "files": [
+                        {
+                            "name": "result.txt",
+                            "mtime": "2021-05-21T24:00:00",
+                            "size": 0,
+                            "mode": "0o644",
+                            "type": "reg",
+                        },
+                        {
+                            "name": "user-benchmark.cmd",
+                            "mtime": "2021-05-21T24:00:00",
+                            "size": 114,
+                            "mode": "0o755",
+                            "type": "reg",
+                        },
+                    ],
+                    "run_data_parent": "ece030b7b37ajsdgfa66s6a7a6be804",
+                    "authorization": {"owner": "1", "access": "private"},
+                    "@timestamp": "2021-05-21T24:00:00.141295",
+                },
+            }
+
+        Output:
+
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
```

Comment:
While I'm hesitant to suggest expanding the Elasticsearch example to include the `sample1` directory document as well, which is bulky and awkward, that would show why this entry appears; which right now might be a bit of a mystery to a casual reader.

Reply:
Yes, this example is confusing.

First, I think that the text should make it clear that this is the contents of the `es_json` parameter...and, as it stands, I think this is just one "hit", not the whole thing that this function will receive.

Second, I don't see a `sample1` subdirectory in the ES response.  That would be a second hit, which isn't shown.  I concur with Dave's assertion that it will be bulky and awkward, but I think it will be worthwhile.